tests/units/common/cleaning/test_clean_whitespace_characters.py

```diff
@@ -0,0 +1,19 @@
+from common.cleaning.clean_whitespace_characters import clean_whitespace_characters
+
+expected_string = 'Simple input with spaces'
+string = ' Simple input with spaces '
+string_with_double_spaces = ' Simple   input   with    spaces '
+string_with_new_lines = ' Simple   input  \n  with  \n spaces '
+
+
+def test_clean_whitespace_characters():
+    output = clean_whitespace_characters(string)
+    assert output == expected_string
+
+def test_clean_whitespace_characters_with_double_spaces():
+    output = clean_whitespace_characters(string_with_double_spaces)
+    assert output == expected_string
+
+def test_clean_whitespace_characters_with_new_lines():
+    output = clean_whitespace_characters(string_with_new_lines)
+    assert output == expected_string
```

Comment:
You could use https://docs.pytest.org/en/6.2.x/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions not to repeat the same test :)

Reply:
I like the suggestion of @drjova , because this test is simple, and does precisely the same thing: compare two strings.  If at least one test fails, we will see the input and expected output in the error

Reply:
`parametrise` is the way to go for such cases. It will give you the info to understand which case has the problem by displaying each case as [a separate test](https://docs.pytest.org/en/6.2.x/example/parametrize.html#generating-parameters-combinations-depending-on-command-line). Additionally, you can add new use cases without writing a new test. At the same time, if the function's name changes, you only have to do it in one place and not in `x` places.

Reply:
Ohh yes I see ! It's good that we can put custom names (ids).
Do you mind [adding test ids (names)](https://docs.pytest.org/en/6.2.x/example/parametrize.html#different-options-for-test-ids) to have the best of both approaches ?